request.go

```diff
@@ -416,9 +416,12 @@ func unmarshalAttribute(
 		return
 	}
 
-	// JSON value was a float (numeric)
-	if value.Kind() == reflect.Float64 {
-		value, err = handleNumeric(attribute, fieldType, fieldValue)
+	switch attribute.(type) {
+	case json.Number:
+		value, err = handleNumeric(attribute.(json.Number), fieldType, fieldValue)
+		return
+	case float64:
```

Comment:
While I think we shouldn't need this case,I had to add it, otherwise the two nested struct tests would fail.

Reply:
Today I had some free time to look into this. I realised that any numeric type gets converted to a `float64` in [handleStruct()](https://github.com/google/jsonapi/blob/master/request.go#L610).

We'll have to ensure the numeric precision isn't lost in the [unmarshal](https://github.com/google/jsonapi/blob/master/request.go#L620) call as well.

Reply:
This has now been sorted in https://github.com/google/jsonapi/pull/206/commits/7ea5e24053020109dfff6012e9241696b2682c33